rest/data/buildlogger.go

```diff
@@ -260,20 +260,21 @@ func (dbc *DBConnector) FindGroupedLogs(ctx context.Context, opts BuildloggerOpt
 
 	opts.ProcessName = ""
 	its := []dbModel.LogIterator{}
-	it, err := dbc.findLogsByTestName(ctx, &opts)
-	if err != nil {
-		return data, time.Time{}, paginated, err
+	if opts.TestName != "" {
+		it, err := dbc.findLogsByTestName(ctx, &opts)
+		if err != nil {
+			return data, time.Time{}, paginated, err
+		}
+		its = append(its, it)
+		opts.EmptyTestName = true
+		// Need to set this to false since the last call to findLogsByTestName
+		// has found the latest execution.
+		opts.EmptyExecution = false
 	}
-	its = append(its, it)
-
-	opts.TestName = ""
-	// Need to set this to false since the last call to findLogsByTestName
-	// has found the latest execution.
-	opts.EmptyExecution = false
-	it, err = dbc.findLogsByTestName(ctx, &opts)
+	it, err := dbc.findLogsByTestName(ctx, &opts)
 	if err == nil {
 		its = append(its, it)
-	} else if errResp, ok := err.(gimlet.ErrorResponse); !ok || errResp.StatusCode != http.StatusNotFound {
+	} else if errResp, ok := err.(gimlet.ErrorResponse); !ok || (opts.TestName == "" || errResp.StatusCode != http.StatusNotFound) {
```

Comment:
I don't think the parentheses are necessary, unless I'm missing something obvious. A || B || C should be equivalent to A || (B || C), since ==/!= has higher precedence than ||.

Reply:
You are right!